front/lib/resources/conversation_resource.ts

```diff
@@ -59,19 +67,54 @@ export class ConversationResource extends BaseResource<ConversationModel> {
   static model: ModelStaticWorkspaceAware<ConversationModel> =
     ConversationModel;
 
+  constructor(
+    model: ModelStaticWorkspaceAware<ConversationModel>,
+    blob: Attributes<ConversationModel>,
+    private readonly _space: SpaceResource | null
+  ) {
+    super(ConversationModel, blob);
+  }
+
+  get space(): SpaceResource | null {
+    if (this.spaceId && !this._space) {
+      throw new Error(
+        "This conversation is associated with a space but the related space is not loaded. Action: make sure to load the space when fetching the conversation."
+      );
+    }
+    return this._space;
+  }
+
   static async makeNew(
     auth: Authenticator,
-    blob: Omit<CreationAttributes<ConversationModel>, "workspaceId">
+    blob: Omit<CreationAttributes<ConversationModel>, "workspaceId">,
+    space: SpaceResource | null
   ): Promise<ConversationResource> {
     const workspace = auth.getNonNullableWorkspace();
+
+    // Check if the user has access to the space.
+    // Note, using canRead because spaces members do not have write access to the space as write is tied with datasources.
+    if (space && !space.canRead(auth)) {
+      throw new Error(
+        "Cannot create conversation in a space you do not have access to."
+      );
+    }
+
+    // Check if the space match the workspace.
+    if (space && space.workspaceId !== workspace.id) {
```

Comment:
I would expect the `space.canRead(auth)` to already handle this case. Maybe worth a refactoring on `canRead` later?